### PR TITLE
[Datasets] Update annotated.txt with corrected annotations

### PR DESCRIPTION
**Pull request by Rebecca-Qian**
# Description

Follow up from #347 , with @aszlam 's annotation corrections. There are 22 annotations updated here, mostly correcting location/selectors.

Also updated JSON schema to enable this:
- Add `ALL` to selectors
- Add `OTHER` to location

Fixes # (issue)

## Type of change

Please check the options that are relevant.

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] Proposes a change (non-breaking change that isn't necessarily a bug)
- [ ] Refactor
- [ ] New feature (non-breaking change that adds a new functionality)
- [ ] Breaking change (fix or feature that would break some existing functionality downstream)
- [ ] This is a unit test
- [ ] Documentation only change
- [x] Datasets Release
- [ ] Models Release

## Type of requested review

- [ ] I want a thorough review of the implementation.
- [ ] I want a high level review. 
- [x] I want a deep design review.

# Testing

Please describe the tests that you ran to verify your changes and how we can reproduce.

# Checklist:

- [x] I have performed manual end-to-end testing of the feature in my environment.
- [ ] I have added Docstrings and comments to the code.
- [ ] I have made changes to existing documentation where needed.
- [ ] I have added tests that show that the PR is functional.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have added relevant collaborators to review the PR before merge.

### CODE REVIEW

**Comment by kavyasrinet** [comment]
Can you also update the spec files with the changes mentioned ? I don't see those in the markdowns.

**Comment by Rebecca-Qian** [comment]
> Can you also update the spec files with the changes mentioned ? I don't see those in the markdowns.

Updated, can you review again?

**Review by kavyasrinet — APPROVED** [review]
Lets merge this but after Arthur gets back we should discuss whether "selector" can have "ALL" as a value. IMO we should have `"selector" : {"return_quantity" : "ALL"}` but don't want to have this PR blocked on that

**Pull request merged**
